### PR TITLE
osemgrep: cleanup Metrics_.ml

### PR DESCRIPTION
**Pull request by aryx**
test plan:
make core

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: src/osemgrep/core/Metrics_.ml [hunk 1]
```diff
@@ -365,13 +351,10 @@ let add_profiling profiler =
...
-
 let add_exit_code code =
   let code = Exit_code.to_int code in
   g.payload.errors.returnCode <- Some code
 
 let add_feature ~category ~name =
   let str = Format.asprintf "%s/%s" category name in
-  g.payload.value.features <- str :: g.payload.value.features;
-  g.payload.value.features <- List.sort String.compare g.payload.value.features
+  g.payload.value.features <- str :: g.payload.value.features
```
Comment (zzeleznick) [inline src/osemgrep/core/Metrics_.ml:360]: note: The python implementation sorts the feature tags. Personally, I don't think this is necessary, but we might expect some random test failures if the features field in the metrics output is not sorted. Perhaps a comment will do?

File: src/osemgrep/core/Metrics_.ml [hunk 2]
```diff
@@ -54,43 +50,48 @@ open Unix
...
 (* Types and constants *)
 (*****************************************************************************)
 
-let _metrics_endpoint = "https://metrics.semgrep.dev"
-let _version = spf "%s" Version.version
-let _base_user_agent = spf "Semgrep/%s" _version
+(* TODO: set to the cannonical "metrics.semgrep.dev" once we upgrade the
+ * host from TLS 1.2 to TLS 1.3
+ *)
+let metrics_url =
```
Comment (zzeleznick) [inline src/osemgrep/core/Metrics_.ml:83]: nice! glad we have this metrics url exposed properly now.

File: src/osemgrep/core/Metrics_.mli [hunk 3]
```diff
@@ -1,55 +1,104 @@
...
- * below and finally accessed in send() further below.
- *)
+(* set g.config *)
 val configure : config -> unit
-val add_engine_type : name:string -> unit
-val is_using_registry : unit -> bool
+
+(* check whether g.config is On (or Auto) *)
 val is_enabled : unit -> bool
-val set_is_using_registry : is_using_registry:bool -> unit
```
Comment (zzeleznick) [inline src/osemgrep/core/Metrics_.mli:32]: note: think we do some overrides for the metrics config if we use the registry. assuming we haven't implemented yet.

**Review by zzeleznick — APPROVED** [review]
Left a few comments for knowledge sharing / context unpacking that can be resolved or turned into code comments

File: src/osemgrep/core/Metrics_.ml (hunk 1 above)
Comment (aryx) [reply src/osemgrep/core/Metrics_.ml:360]: as soon as we get failures I'll add the sorting code and comments.

File: src/osemgrep/core/Metrics_.mli (hunk 3 above)
Comment (aryx) [reply src/osemgrep/core/Metrics_.mli:32]: yes I've added a TODO comment actually in the toplevel comment. I think that's the main thing missing with
the parsing data and more profiling.

**Pull request merged**